### DISCUSSION TITLE
Cache Details (TTL? LRU?)

### DISCUSSION THREAD

**Discussion by RobertAron** [vercel/swr#2750] [Q&A] [answered]
I'm slightly worried about my application memory performance. We have a large number of keys used. I think I want to implement a TTL cache, but before doing that I want to insure that doing that kind of customization is even necessary.

I read this but it isn't really clear what the end result was.
https://github.com/vercel/swr/pull/92#issuecomment-603547557

What is the default implementation? Does the cache grow infinitely? Does it ever throw away keys if the size cache becomes large?

**Comment by msdrigg** [comment, marked as answer]
I was looking this over for myself (user, not maintainer), and from what I can tell the cache grows indefinitely. 

I came to this conclusion because the default cache provider is a plain js `Map` object (https://github.com/vercel/swr/blob/e510955d42b472c8d45034c51b8acc12980aa2e7/_internal/src/utils/config.ts#L44). So unless there is some magic under the hood that I can't see, there is currently no built in support for TTL, LRU or anything. You would need to use a SwrConfigProvider and supply your own cache.
